src/library/scala/collection/mutable/ListBuffer.scala

```diff
@@ -120,31 +120,35 @@ class ListBuffer[A]
     this
   }
 
-  // Overridden for performance
-  override final def addAll(xs: IterableOnce[A]): this.type = {
-    if (xs.asInstanceOf[AnyRef] eq this) { // avoid mutating under our own iterator
-      if (len > 0) {
-        ensureUnaliased()
-        val copy = ListBuffer.from(this)
-        last0.next = copy.first
-        last0 = copy.last0
-        len *= 2
-      }
-    } else {
-      val it = xs.iterator
-      if (it.hasNext) {
-        ensureUnaliased()
+  // MUST only be called on fresh instances
```

Comment:
If the only possible use is to call it on a fresh instance, why not create the instance too?  Otherwise, doesn't it just expose a risk of mis-use while requiring more boilerplate at the call site?